### PR TITLE
Faster, more space efficient and simpler solution to "Subsequences Summing to Sevens"

### PR DESCRIPTION
**Pull request by thetazero**
The solution passes all test cases and is more straight forward than the current solution. 
It uses simpler data structures and eliminates redundant writes to the prefix sum array.
Hope this helps!

### CODE REVIEW

**Comment by thecodingwizard** [comment]
I think this should be provided as an alternate solution (along with an explanation for how it works) rather than replacing the current solution? The problem is part of the prefix sums module so it's nice for at least one implementation to explicitly use prefix sums.

**Comment by thetazero** [comment]
Yeah, I considered that however, the current solution seems not to make use of prefix sums, I believe it just generates the array. It exclusively uses pref[i + 1] which seems to be a roundabout equivalent to just having an int. However, I might be wrong. 

I can either do what you suggested or perhaps this problem should be moved from the prefix sums module. Regardless, I'll add an explanation.
Thoughts?

**Comment by frodakcin** [comment]
Yea honestly the original solution isn't very good 😐

**Comment by Farrius** [comment]
Yes, you are right pref[i + 1] is equivalent to have an int, and your solution is way better. However I would  replace  cur for a "prefix sum" array in order to use prefix sums in some way.

**Comment by frodakcin** [comment]
Sorry, I sounded very rude. To elaborate, I thought using the pref array was fine. The only thing I dislike about the soln was the use of sets and maps, which could be so easily avoided.

**Comment by thetazero** [comment]
Changed the solution to use a prefix array as suggested.

**Comment by thetazero** [comment]
Sorry for a second commit, I realized I broke a link

File: solutions/USACO jan16-subseq-summing-to-sevens.mdx [hunk 1]
```diff
@@ -5,7 +5,7 @@ title: Subsequences Summing to Sevens
 author: Óscar Garries
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_div7_silver_jan16.html)
+[Official Analysis](http://www.usaco.org/currrent/data/sol_div7_silver_jan16.html)
```
Comment (dongliu0426) [inline solutions/USACO jan16-subseq-summing-to-sevens.mdx:8]: typo: "currrent" should be "current"

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/USACO jan16-subseq-summing-to-sevens.mdx (hunk 1 above)
Comment (thetazero) [reply solutions/USACO jan16-subseq-summing-to-sevens.mdx:8]: thanks! can't believe I missed that

Comment (bqi343) [reply solutions/USACO jan16-subseq-summing-to-sevens.mdx:8]: > thanks! can't believe I missed that

please fix this (and also update the author)

**Pull request merged**